src/portable/portable.c

```diff
@@ -1425,21 +1425,32 @@ static bool marker_matches_images(const char *marker, const char *name_or_path,
                         if(!(e[strspn(e, "/")] == 0 || streq(e, ".raw")))
                                 return false;
                 } else {
-                        const char *b, *underscore;
+                        const char *b, *e, *underscore;
                         size_t l;
 
                         /* We shall match against a path. Let's ignore any prefix here though, as often there are many ways to
-                        * reach the same file. However, in this mode, let's validate any file suffix. */
+                         * reach the same file. However, in this mode, let's validate any file suffix. But also ensure that we
+                         * don't fail if both components don't have a '/' at all (strcspn returns the full length of the string
+                         * in that case, which might not match as the versions might differ). */
 
+                        e = strchr(a, '/');
                         l = strcspn(a, "/");
```

Comment:
these two lines to some level do the same thing but one returns a pointer, the other a length. one of them should go or be replaced by the other.

Reply:
i.e.:

```c
l = e ? e - a : strlen(a);
```